src/libopenrave-core/environment-core.h

```diff
@@ -3002,14 +3002,14 @@ class Environment : public EnvironmentBase
                 }
 
                 if( !bChanged ) {
-                    dReal dist = TransformDistanceFast(pInitBody->GetTransform(), pKinBodyInfo->_transform);
+                    dReal dist = TransformDistanceFast(pInitBody->GetTransform(), pKinBodyInfo->_transform * pInitBody->_baseLinkRelativeTransform);
```

Comment:
err... doesn't make sense, shouldn't `pKinBodyInfo->_transform` be the body frame? So not sure why you have to change here

Reply:
Originally, I was trying to preserve `GetLinks()[0]->GetTransform() == GetTransform()`. So I needed to make `SetTransform` account for `body in world` + `base link in body`.